src/syntax/herbie-to-egg-pattern.rkt

```diff
@@ -0,0 +1,86 @@
+#lang racket
+(require "./rules.rkt")
+(provide herbie-pattern->rust-pattern)
+
+
+(define (herbie-pattern->rust-pattern datum)
+  (cond
+    [(list? datum)
+     (string-append
+      "("
+      (symbol->string (first datum))
+     (foldr
+      (lambda (sub-expr acc)
+        (string-append " "
+                       (herbie-pattern->rust-pattern sub-expr)
+                       acc))
+      ""
+      (rest datum))
+     ")")]
+    [(symbol? datum)
+     (if
+      (set-member? fpconstants datum)
+      (symbol->string datum)
+      (string-append "?" (symbol->string datum)))]
+    [(number? datum)
+     (number->string datum)]
+    [else
+     (error "expected list, number, or symbol")]))
+
+;; use as a command line tuool  to generate rules in the egg rust syntax
+;; this is good for direct use on the rust side (testing)
+(module+ main
```

Comment:
I get that this is debugging code so I'm fine merging as is, but there are two simple changes you can make to simplify this a lot.

1) Instead of reading `rules.rkt` as a syntax object and filtering the groups, just use `(*simplify-rules*)`
2) Use `format` and `string-join` instead of `string-append` and `folds`